### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by m-talbi**
- Populated database with seeds.
- Added associations between models.
- Added methods to each model.

### CODE REVIEW

**Comment by codecaiine** [comment]
Hi @Kweeka1 can you go on slack ? I left a message there

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :user, foreign_key: :author_id
+
```
Comment (codecaiine) [inline app/models/post.rb:6]: - [ ] [OPTIONAL] I would like to recommend you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private :+1:

**Review by codecaiine — APPROVED** [review]
Hi @Kweeka1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- The associations between models are set up correctly :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (m-talbi) [reply app/models/post.rb:6]: that's a cool feature. Thanks @codecaiine.

**Pull request merged**